Incompatibilities with xarray > 2022.03

### Expected behaviour
ODC should work with current version of `xarray`. In `setup.py` there's an exclusion of `2022.6.0`, but I don't think that's sufficient. It'd be worth digging up the commit/PR that made that change.

### Actual behaviour

Tests are failing.

```
FAILED tests/api/test_grid_workflow.py::test_gridworkflow_with_time_depth - AssertionError
FAILED tests/api/test_virtual.py::test_aggregate - ValueError: time already exists as coordinate or variable name.
```

### Steps to reproduce the behaviour

`pytest tests/`

### Environment information

* Which ``datacube --version`` are you using?

 `develop` branch at `af59377327c363b9c52b55000b4024a0b3fbaa8b`

* What datacube deployment/enviornment are you running against?

  - Mambaforge
  - conda-forge 
  - Python 3.10

--- a/setup.py
+++ b/setup.py
@@ -106,7 +106,7 @@ setup(
         'sqlalchemy',
         'GeoAlchemy2',
         'toolz',
-        'xarray>=0.9,!=2022.6.0',  # >0.9 fixes most problems with `crs` attributes being lost
+        'xarray>=0.9,<2022.6',  # >0.9 fixes most problems with `crs` attributes being lost
     ],
     extras_require=extras_require,
     tests_require=tests_require,

